sources/api/thar-be-updates/src/main.rs

```diff
@@ -0,0 +1,562 @@
+    let status_file = File::open(UPDATE_STATUS_FILE).context(error::OpenStatusFile {
+        path: UPDATE_STATUS_FILE,
+    })?;
+    Ok(serde_json::from_reader(status_file).context(error::StatusParse)?)
+}
+
+/// Spawns updog process to get list of updates and check if any of them can be updated to.
+/// Returns true if there is an available update, returns false otherwise.
+fn refresh_update(
+    status: &mut UpdateStatus<update_metadata::Update>,
+    socket_path: &str,
+) -> Result<bool> {
+    debug!("Spawning 'updog whats'");
+    let output = Command::new("updog")
+        .args(&["whats", "--all", "--json"])
+        .output()
+        .context(error::Updog)?;
+    status.set_recent_command_info(UpdateCommand::RefreshUpdate, Some(&output), None);
+    if output.status.success() {
+        let update_info: Vec<update_metadata::Update> =
+            serde_json::from_slice(&output.stdout).context(error::UpdateInfo)?;
+        return status.update_available_updates(socket_path, update_info);
+    } else {
+        warn!("Failed to check for updates with updog");
+    }
+    Ok(false)
+}
+
+/// Queries the Bottlerocket API and returns the 'ignore-waves' setting. If it doesn't exist for whatever reason, return false
+fn get_ignore_waves(socket_path: &str) -> Result<bool> {
+    let uri = "/settings";
+    let method = "GET";
+    let (code, response_body) = apiclient::raw_request(&socket_path, uri, method, None)
+        .context(error::APIRequest { method, uri })?;
+    ensure!(
+        code.is_success(),
+        error::APIResponse {
+            method,
+            uri,
+            code,
+            response_body,
+        }
+    );
+    let settings: serde_json::Value =
+        serde_json::from_str(&response_body).context(error::ResponseJson { uri })?;
+    Ok(settings["updates"]["ignore-waves"]
+        .as_bool()
+        .unwrap_or(false))
+}
+
+/// Prepares the update by downloading and writing the update to the staging partition
+fn prepare_update(
+    status: &mut UpdateStatus<update_metadata::Update>,
+    ignore_waves: bool,
+) -> Result<()> {
+    debug!("Spawning 'updog update-image'");
+    let chosen_update = status
+        .get_chosen_update()
+        .context(error::ChosenUpdate)?
+        .clone();
+    // Prepare arguments, append '--now' if we're ignoring waves
+    let chosen_version = chosen_update.get_version().to_string();
+    let mut args = vec!["update-image", "--image", &chosen_version];
+    if ignore_waves {
+        args.push("--now");
+    }
+    let output = Command::new("updog")
+        .args(args)
+        .output()
+        .context(error::Updog)?;
+    status.set_recent_command_info(UpdateCommand::PrepareUpdate, Some(&output), None);
+    if output.status.success() {
+        status.set_staging_partition_image_info(chosen_update);
+    } else {
+        warn!("Failed to prepare the update with updog");
+    }
+    Ok(())
+}
```

Comment:
I would expect `updog` to retrieve and understand the relevant API settings (`version-lock` and `ignore-waves`) so that we don't have divergent behavior if it's run as a standalone tool.

It can accept arguments on the command-line to override, but we shouldn't need to override them here.

If we do that, then we can refactor `refresh`, `prepare`, `activate`, and `deactivate` to share the same execution logic.

Reply:
> I would expect updog to retrieve and understand the relevant API settings

The idea of thar-be-updates is that it's the piece that understands the API, the state machine, and does the locking, and updog can be the normal CLI experience separate from the API.  I think that separation is valuable; updog is doing enough, IMO.  It can remain the manual override, for when an escape from the API is needed.

Reply:
We can put the desired version and the ignore waves flag in `/etc/updog.toml`.

I'd like for `updog update` to work the same as an update through the API, and leave overrides to happen through the command line.

Reply:
+1; even with the separation I described, I think we could put the new version-request setting in updog's config template, and updog could respect that setting by default, and still allow `-i` to override version.